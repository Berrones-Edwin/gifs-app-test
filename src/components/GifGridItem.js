import Proptypes from 'prop-types'
import React from 'react'

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <p> {title} </p>
    </div>
  )
}

GifGridItem.prototypes = {
  title: Proptypes.string.isRequired,
  url: Proptypes.string.isRequired
}
