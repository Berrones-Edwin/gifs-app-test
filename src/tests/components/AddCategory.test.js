import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import React from 'react'
import { AddCategory } from '../../components/AddCategory'
describe('test in component <AddCAtegory />', () => {
  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />)

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test('should show component successfully', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should change el value in input', () => {
    const inputText = wrapper.find('input')

    const value = 'hello world'

    inputText.simulate('change', {
      target: {
        value
      }
    })
    const paragraph = wrapper.find('p').text().trim()
    expect(paragraph).toBe(value)
  })
  test('should not send the info', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} })
    expect(setCategories).not.toHaveBeenCalled()
  })

  test('should call setCategories method and clean input txt', () => {
    const value = 'hola mundo'
    wrapper.find('input').simulate('change', {
      target: { value }
    })

    wrapper.find('form').simulate('submit', { preventDefault() {} })

    expect(setCategories).toHaveBeenCalled()
    expect(setCategories).toHaveBeenCalledTimes(1) //se haya llamando una vez
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function))

    expect(wrapper.find('input').props().value).toBe('')
  })
})
