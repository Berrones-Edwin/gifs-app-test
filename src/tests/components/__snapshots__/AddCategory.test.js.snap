// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in component <AddCAtegory /> should show component successfully 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
